src/ApplePayGateway.php

```diff
@@ -0,0 +1,206 @@
+ *    $gateway->setUsername('your_username');
+ *    $gateway->setPassword('y0ur_p4ssw0rd');
+ *    $gateway->setPemCertPath('./certs/path_to_cert.crt.pem');
+ *    $gateway->setKeyCertPath('./certs/path_to_cert.key.pem');
+ *    $gateway->setKeyCertPassword('your_key_cert_password');
+ *    $gateway->setTestMode(false);
+ *
+ *    // User clicks or taps an Apple Pay button. The payment sheet is partially loaded. So now Apple
+ *    // can validate your merchant identity and you can receive a session object to fully load the
+ *    // payment sheet to accept payment.
+ *
+ *    // To request the payment session pass the required Apple Pay options to the authorize method
+ *    // and send the request. An AbstractResponse object will be returned.
+ *    $authorizeResponse = $gateway->authorize(array(
+ *        // sandbox validation Url
+ *        'validationURL' => 'https://apple-pay-gateway-cert.apple.com/paymentservices/startSession',
+ *        'merchantIdentifier' => 'merchant.com.example',
+ *        'displayName' => 'MyStore',
+ *        'applicationUrl' => 'mystore.com' // initiativeContext: the domain name of your site.
+ *    ))->send();
+ *
+ *    // getApplePayPaymentSessionObject() can then be called to retrieve the ApplePayPaymentSession object.
+ *    // You can ensure the response is successful before sending it to the front end. If there is an error the
+ *    // response body will be empty.
+ *    if ($authorizeResponse->isSuccessful()) {
+ *        echo 'Status Code: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Status Reason: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *        echo 'Apple Pay Payment Session Object: ' . $authorizeResponse->getPaymentSessionObject()
+ *          . PHP_EOL;
+ *        echo 'Response: ' . $authorizeResponse->getResponse() . PHP_EOL;
+ *    } else {
+ *        // Error handling.
+ *        echo 'ERROR: Request not successful.';
+ *        echo 'Status Code: ' . $authorizeResponse->getCode() . PHP_EOL;
+ *        echo 'Status Message: ' . $authorizeResponse->getMessage() . PHP_EOL;
+ *    }
+ *
+ *    //Pass authorizeResponse back to the client to validate your merchant and continue with an Apple Pay payment.
+ *    //If successful, the payment sheet should be fully loaded.
+ *    $apple_pay_session = $authorizeResponse->getPaymentSessionObject();
+ *
+ *     // TODO: Add functionality for completeAuthorize() and capture().
+ *     // An opaque Apple Pay Session is returned as a response (expires after 5 mins) and it can be sent to
+ *     // the front end to fully load the payment sheet. This allows the user to optionally configure their
+ *     // payment option and shipping methods (if needed) and submit their payment.
+ * </code>
+ */
+class ApplePayGateway extends AbstractVindiciaGateway
+{
+
+    /**
+     * Get the gateway name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return 'Vindicia ApplePay';
```

Comment:
This could be a constant as well?

Reply:
Don't think this is necessary since it's only instance is here.

Reply:
works for me, this is how Omnipay typically does it